### PR TITLE
DAOS-4307 sec: Check permissions on container reads

### PR DESCRIPTION
**Pull request by dpquigl**
Previously the assumption was a container handle had RO
or RW access, with no further checking. Now with granular
ACL permissions, a user with access to properties or
administrative functions may not be allowed to read the
actual data.

- Add the ds_sec_cont_can_read_data function to verify
  that the capabilities include data read.

Signed-off-by: Kris Jacque <kristin.jacque@intel.com>
Signed-off-by: David Quigley <david.quigley@intel.com>

### CODE REVIEW

File: src/object/srv_obj.c [hunk 1]
```diff
@@ -1050,6 +1050,16 @@ obj_ioc_init(uuid_t pool_uuid, uuid_t coh_uuid, uuid_t cont_uuid, int opc,
 		return rc;
 	}
 
+	// TODO: Needs to be cleaned up maybe pulling all checks into their own
+	// function.
```
Comment (kjacque) [inline src/object/srv_obj.c]: I think the check looks OK as-is. The comment can be removed entirely.

File: src/security/srv_acl.c [hunk 2]
```diff
@@ -758,3 +758,12 @@ ds_sec_cont_can_read_data(uint64_t cont_capas)
...
 	return (cont_capas & CONT_CAPA_READ_DATA) != 0;
 }
+
+uint64_t
+ds_sec_get_rebuild_cont_capabilities(void)
+{
+	/*
+	 * Internally generated rebuild container handles can read data
+	 */
+	return CONT_CAPA_READ_DATA;
```
Comment (dpquigl) [inline src/security/srv_acl.c]: Is this only read or do we need write as well? Also does it get or set any props?

Comment (kjacque) [reply src/security/srv_acl.c]: I believe (based on some discussions) that rebuild only goes through code paths shared with the client for the object read. That would explain why rebuild tests didn't start failing until we started checking for read permissions here. Get and set prop permissions are checked pretty high up in the client entry point.

Comment (kjacque) [reply src/security/srv_acl.c]: Looks like we've passed Functional tests, which included the rebuild cases we were failing before.

**Comment by kjacque** [comment]
The Functional test failures were both [DAOS-3955](https://jira.hpdd.intel.com/browse/DAOS-3955).

**Pull request merged**